WF-30 [back] User test

guzus opened the pull request:
User Test 입니다. 
Social Login 테스트도 하고 싶은데 이게 가능한지 모르겠네요  
test는 풍성할수록 좋을 것 같으니까 의견 많이 주세요!

guzus committed:
init

need improvements

guzus committed:
Merge remote-tracking branch 'origin/main' into feature/WF-30

guzus committed:
fix error

Test is weak yet, but will be stronger soon!

guzus committed:
Merge remote-tracking branch 'origin/main' into feature/WF-30

guzus committed:
Update tests.py

YeonghyeonKO commented on wafflow/user/tests.py:
```diff
@@ -1,3 +1,517 @@
+        self.check_guzus(data)
+
+        # response = self.client.get(
+        #     f"/user/{self.guzus.id}/",
```
이부분은 왜 주석인가요??

eldpswp99 replied:
get /user/{id}/ 에서 invalid token일 때도 get 할 수 없는 오류를 발견했습니다. 근데 이건 굳이 건드리지 않아도 상관없을 것 같아서 일단 두었습니다.

이부분인듯합니다

guzus replied:
네 명훈님 말이 맞습니다!

YeonghyeonKO commented on wafflow/user/tests.py:
```diff
@@ -1,3 +1,517 @@
+
+    def test_put_user_me_invalid_pk(self):
+        response = self.client.put(
+            "/user/you/",
```
url 잘못 따고 들어간 케이스인거죠?

guzus replied:
그렇습니다! 이럴 일은 거의 없겠지만..

YeonghyeonKO reviewed (approved):
수고하셨습니다!

eldpswp99 commented on wafflow/user/tests.py:
```diff
@@ -1,3 +1,517 @@
+        # self.assertEqual(response.status_code, status.HTTP_200_OK)
+        # data = response.json()
+        # self.check_guzus(data)
+
```
question, answer, bookmark 모델을 활용하여 객체를 만들어서 answer,question,bookmark count가 잘 작동한는지도 확인이 필요해 보입니다!

guzus replied:
넵 그부분을 넣어야되나 고민했는데 역시 넣는 것이 맞겠죠?

eldpswp99 replied:
네네

eldpswp99 commented on wafflow/user/tests.py:
```diff
@@ -1,3 +1,517 @@
+    def test_put_user_me_too_long_nickname(self):
+        response = self.client.put(
+            "/user/me/",
+            json.dumps({"password": "12345", "nickname": "Django" * 10}),
```
닉네임이 16자 제한인데 17자 정도로 요청 넣어서 판단해주시면 좀 더 엄밀하게 테스트가 가능할 듯 싶어요 지금은 너무 길어서 좀 더 강도를 높일 수 있을 듯 합니다

eldpswp99 commented on wafflow/user/tests.py:
```diff
@@ -1,3 +1,517 @@
+        data = response.json()
+        self.check_guzus(data)
+        self.assertIn("token", data)
+
```
16자 이상의 username을 이용해 로그인하려는 시도에 대한 validation로직이 있어야 할 것 같습니다. 깃허브 소셜 로그인 유저가 username 로그인을 못하게 하기 위한 중요한 validation이라서요

eldpswp99 commented on wafflow/user/tests.py:
```diff
@@ -1,3 +1,517 @@
+            content_type="application/json",
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
```
생각해보면, 지금 logout함수는 어떠한 역할도 하지 않는 것처럼 느껴집니다 로그아웃을 해도 그대로 토큰을 사용해 로그인할수 있으니까요. 지금의 토큰은 expired되지도 않아서 너무 보안적으로 취약하다고 생각되네요. logout될 때, 기존의 token object를 삭제하는 로직, 그리고 token expire과 관련된 로직을 추가하는 것이 필요하다고 느껴집니다. 지금은 body로 아이디와 패스워드를 모두 보내는 상황이여서.. 아니면 session login으로 바꾸는 것도 좋구요

YeonghyeonKO replied:
slack 관련 링크 : https://wafflestudio1-l3f6523.slack.com/archives/C01ELEGPSBG/p1609479955267000

guzus committed:
Merge remote-tracking branch 'origin/main' into feature/WF-30

guzus committed:
update

login with long username, more strict nickname validuation, question_count answer_count bookmark_count check

guzus committed:
is_active

small limprovement about is_active

guzus committed:
ci

guzus committed:
Update tests.py
